setup-slave.sh

```diff
@@ -21,12 +21,15 @@ HOSTNAME=$PRIVATE_DNS  # Fix the bash built-in hostname variable too
 
 echo "Setting up slave on `hostname`..."
 
-# Work around for R3 instances without pre-formatted ext3 disks
+# Mount options to use for ext3 and xfs disks (the ephemeral disks
+# are ext3, but we use xfs for EBS volumes to format them faster)
+XFS_MOUNT_OPTS="defaults,noatime,nodiratime,allocsize=8m"
+# ext4 has the best performance among ext3, ext4, and xfs based on our shuffle heavy benchmark
+EXT4_MOUNT_OPTS="defaults,noatime,nodiratime"
+
 instance_type=$(curl http://169.254.169.254/latest/meta-data/instance-type 2> /dev/null)
-if [[ $instance_type == r3* ]]; then
-  # Format & mount using ext4, which has the best performance among ext3, ext4, and xfs based
-  # on our shuffle heavy benchmark
-  EXT4_MOUNT_OPTS="defaults,noatime,nodiratime"
+if [[ $instance_type == r3* || $instance_type == i2* ]]; then
+  # Work around for R3 and I2 instances without pre-formatted and pre-mounted ext3 disks.
```

Comment:
Why is this a workaround? Looks like this comment was copied from the upstream setup-slave.sh, but it doesn't apply here, since for us not having a pre-mounted ext3 disk is a good thing!